packages/babel-core/src/transformation/file/index.js

```diff
@@ -359,21 +359,28 @@ export default class File extends Store {
         sourceRoot: inputMapConsumer.sourceRoot
       });
 
+      // This assumes the output map always has a single source, since Babel always compiles a single source file to a
+      // single output file.
+      const source = outputMapConsumer.sources[0];
```

Comment:
The code is correct.  `source` is being passed to `outputMapConsumer.generatedPositionFor`.  The first argument to that function must describe a position in one of `outputMapConsumer`'s source files, not `inputMapConsumer`'s or anything else.  Hence, we grab `outputMapConsumer.sources[0]`.

`mapping.source` refers to the source file of the *input* map.  However, the output map's source correspond to the input map's *generated code*.